src/app/components/login/login.component.ts

```diff
@@ -1,89 +1,123 @@
-            : (this.isLightThemeEnabled = false);
+    let getCurrentUser = JSON.parse(localStorage.getItem(`isRememberMe`)) ?? null;
+    if (getCurrentUser && getCurrentUser.isRememberMe) {
+      this.loginForm.get('username').setValue(getCurrentUser.username);
+      this.loginForm.get('password').setValue(getCurrentUser.password);
+      this.isRememberMeCheked = getCurrentUser.isRememberMe;
     }
+  }
 
-    public login() {
-        if (this.loginForm.get('username').invalid || this.loginForm.get('password').invalid) {
-            this.toasterService.error("Please enter username and password")
-            return;
-        }
+  public login() {
+    if (this.loginForm.get('username').invalid || this.loginForm.get('password').invalid) {
+      this.toasterService.error('Please enter username and password');
+      return;
+    }
 
-        let username = this.loginForm.get('username').value;
-        let password = this.loginForm.get('password').value;
+    let username = this.loginForm.get('username').value;
+    let password = this.loginForm.get('password').value;
 
-        this.loginService.login(this.server, username, password).subscribe(async (response: AuthResponse) => {
-            let server = this.server;
-            server.authToken = response.access_token;
-            server.username = username;
-            server.password = password;
-            await this.serverService.update(server);
+    this.loginService.login(this.server, username, password).subscribe(
+      async (response: AuthResponse) => {
+        let server = this.server;
+        server.authToken = response.access_token;
+        server.username = username;
+        server.password = password;
+        server.tokenExpired = false;
+        await this.serverService.update(server);
 
-            if (this.returnUrl.length <= 1) {
-                this.router.navigate(['/server', this.server.id, 'projects'])
-            } else {
-                this.router.navigateByUrl(this.returnUrl);
-            }
+        if (this.returnUrl.length <= 1) {
+          this.router.navigate(['/server', this.server.id, 'projects']);
+        } else {
+          this.router.navigateByUrl(this.returnUrl);
+        }
+      },
+      (error) => {
+        this.loginError = true;
+      }
+    );
+  }
+
+  rememberMe(ev) {
+    if (ev.checked) {
+      let curren_user = {
+        username: this.loginForm.get('username').value,
+        password: this.loginForm.get('password').value,
+        isRememberMe: ev.checked,
+      };
+      this.isRememberMeCheked = ev.checked;
+      localStorage.setItem(`isRememberMe`, JSON.stringify(curren_user));
+    } else {
+      localStorage.removeItem(`isRememberMe`);
+      this.loginForm.reset();
+      this.isRememberMe = ev.checked;
+    }
+  }
 
-        }, error => {
-            this.loginError = true;
-        });
+  ngDoCheck() {
+    if (this.loginForm.get('username').valid || this.loginForm.get('password').valid) {
```

Comment:
Should you use a '&&' instead of '||" here? The way I understand this is both username and password should be valid to have the isRememberMe set to true.